File: 1760-minimum-limit-of-balls-in-a-bag/1760-minimum-limit-of-balls-in-a-bag.js
/**
 * @param {number[]} nums
 * @param {number} maxOperations
 * @return {number}
 */
var minimumSize = function(nums, maxOperations) {
  let [low, high] = [1, Math.max(...nums)];
  
  while (low < high) {
    const mid = (low + high) >>> 1;
    const count = getCount(nums, mid);
    if (count > maxOperations) low = mid + 1;
    else high = mid;
  }
  
  return low;
};

const getCount = (nums, mid) => nums.reduce((acc, num) => acc + Math.floor((num - 1) / mid), 0);

File: 1760-minimum-limit-of-balls-in-a-bag/1760-minimum-limit-of-balls-in-a-bag.py
class Solution:
    def minimumSize(self, nums: List[int], maxOperations: int) -> int:
        low, high = 1, max(nums)
        while low < high:
            mid = (low + high) >> 1
            if sum((num - 1) // mid for num in nums) > maxOperations:
                low = mid + 1
            else:
                high = mid
        return low